### PR TITLE
pyparsing: upgrade to 3.1.1

### PR DESCRIPTION
**Pull request by miz060**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Update `pyparsing` to 3.1.1

###### Change Log  

- Update `pyparsing` to 3.1.1

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- BuddyBuild: [504991](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=504991&view=results)

### CODE REVIEW

**Review by mfrw — COMMENTED** [review]
Can we please fill the checklist and attach a test build.

**Comment by miz060** [comment]
> Can we please fill the checklist and attach a test build.

Ready now!

File: SPECS/pyparsing/pyparsing.spec [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 Release:        1%{?dist}
 License:        MIT
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 Group:          Development/Languages/Python
 URL:            https://github.com/pyparsing/pyparsing
-Source0:        https://github.com/pyparsing/pyparsing/releases/download/%{name}_%{version}/%{name}-%{version}.tar.gz
+Source0:        https://github.com/pyparsing/pyparsing/releases/download/%{version}/%{name}-%{version}.tar.gz
 BuildRequires:  python3-devel
-%if %{with_check}
```
Comment (anphel31) [inline SPECS/pyparsing/pyparsing.spec:12]: Are all of these BuildRequires needed even when not running tests?

Comment (miz060) [reply SPECS/pyparsing/pyparsing.spec:12]: Good point! Yes, I verified that those are required for the new build and install steps.

**Pull request merged**